dwiintensitynorm: command failed

Hi all,

Playing around with the `dwiintensitynorm` script but got an error.

Version:
```
== mrview 0.3.14-629-gfb9db785 ==
64 bit release version, built Jun  2 2016, using Eigen 3.2.8
```

Things were going well until:
```
Command: mrmath input_transformed/* mean initial_template.mif
mrmath: [ERROR] mrmath requires either multiple input images, or the -axis option to be provided
population_template: [ERROR] Command failed: mrmath input_transformed/* mean initial_template.mif
population_template: Changing back to original directory (/private/tmp/dwiintensitynorm-tmp-N7OP3R)
population_template: Contents of temporary directory kept, location: /private/tmp/dwiintensitynorm-tmp-N7OP3R/population_template
dwiintensitynorm: [ERROR] Command failed: population_template fa -mask_dir /tmp/dwiintensitynorm-tmp-N7OP3R/mask fa_template.mif -linear_scale 0.25,0.5,1.0,1.0 -nl_scale 0.5,0.75,1.0,1.0,1.0 -nl_niter 5,5,5,5,5 -warp_dir warps -nocleanup -tempdir population_template
dwiintensitynorm: Changing back to original directory (/Users/chyeh/test_data)
dwiintensitynorm: Deleting temporary directory /tmp/dwiintensitynorm-tmp-N7OP3R/
```

@Lestropie Just tried it and got these errors:
```
Command: mrconvert nl_template4.mif /private/tmp/dwiintensitynorm-tmp-POBONS/fa_template.mif
Traceback (most recent call last):
  File "/Users/chyeh/Toolkit/mrtrix3/scripts/population_template", line 363, in <module>
    shutil.copy('warps/', getUserPath(lib.app.args.warp_dir, False))
NameError: name 'shutil' is not defined
dwiintensitynorm: [ERROR] Command failed: population_template fa -mask_dir /tmp/dwiintensitynorm-tmp-POBONS/mask fa_template.mif -linear_scale 0.25,0.5,1.0,1.0 -nl_scale 0.5,0.75,1.0,1.0,1.0 -nl_niter 5,5,5,5,5 -warp_dir warps -nocleanup -tempdir population_template
dwiintensitynorm: Changing back to original directory (/Users/chyeh/test_data)
dwiintensitynorm: Deleting temporary directory /tmp/dwiintensitynorm-tmp-POBONS/
```

```
Traceback (most recent call last):
  File "/Users/chyeh/Toolkit/mrtrix3/scripts/population_template", line 363, in <module>
    shutil.copy('warps/', getUserPath(lib.app.args.warp_dir, False))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 119, in copy
    copyfile(src, dst)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 21] Is a directory: 'warps/'
dwiintensitynorm: [ERROR] Command failed: population_template fa -mask_dir /tmp/dwiintensitynorm-tmp-YUPED6/mask fa_template.mif -linear_scale 0.25,0.5,1.0,1.0 -nl_scale 0.5,0.75,1.0,1.0,1.0 -nl_niter 5,5,5,5,5 -warp_dir warps -nocleanup -tempdir population_template
```

--- a/scripts/lib/app.py
+++ b/scripts/lib/app.py
@@ -169,11 +169,16 @@ def checkOutputFile(path):
   if not path:
     return
   if os.path.exists(path):
+    type = ''
+    if os.path.isfile(path):
+      type = ' file'
+    elif os.path.isdir(path):
+      type = ' directory'
     if args.force:
-      warnMessage('Output file ' + os.path.basename(path) + ' already exists; will be overwritten at script completion')
+      warnMessage('Output' + type + ' ' + os.path.basename(path) + ' already exists; will be overwritten at script completion')
       mrtrixForce = ' -force'
     else:
-      errorMessage('Output file ' + path + ' already exists (use -force to override)')
+      errorMessage('Output' + type + ' ' + path + ' already exists (use -force to override)')
       sys.exit(1)
 
 

--- a/scripts/population_template
+++ b/scripts/population_template
@@ -111,7 +111,10 @@ lib.app.checkOutputFile(lib.app.args.template)
 if lib.app.args.warp_dir:
   lib.app.args.warp_dir = relpath(lib.app.args.warp_dir)
   lib.app.checkOutputFile(lib.app.args.warp_dir)
-  lib.app.make_dir(lib.app.args.warp_dir)
+
+if lib.app.args.transformed_dir:
+  lib.app.args.transformed_dir = relpath(lib.app.args.transformed_dir)
+  lib.app.checkOutputFile(lib.app.args.transformed_dir)
 
 # automatically detect SH series
 do_fod_registration = False
@@ -360,10 +363,16 @@ for level in range(0, len(nl_scales)):
 runCommand('mrconvert ' + current_template + ' ' + getUserPath(lib.app.args.template, True) + lib.app.mrtrixForce)
 
 if lib.app.args.warp_dir:
-  shutil.copy('warps/', getUserPath(lib.app.args.warp_dir, False))
+  warp_path = getUserPath(lib.app.args.warp_dir, False)
+  if os.path.exists(warp_path):
+    shutil.rmtree(warp_path)
+  shutil.copytree('warps/', warp_path)
 
 if lib.app.args.transformed_dir:
-  shutil.copy('input_transformed/', getUserPath(lib.app.args.transformed_dir, False))
+  transformed_path = getUserPath(lib.app.args.transformed_dir, False)
+  if os.path.exists(transformed_path):
+    shutil.rmtree(transformed_path)
+  shutil.copytree('input_transformed/', transformed_path)
 
 if lib.app.args.template_mask:
   runCommand('mrconvert ' + current_template_mask + ' ' + getUserPath(lib.app.args.template_mask, True) + lib.app.mrtrixForce)

--- a/scripts/population_template
+++ b/scripts/population_template
@@ -2,7 +2,7 @@
 
 # Generates an unbiased group-average template via image registration of images to a midway space.
 
-import os, sys
+import os, shutil, sys
 import lib.app
 import numpy
 import math

--- a/scripts/population_template
+++ b/scripts/population_template
@@ -13,6 +13,9 @@ def abspath(*arg):
 def relpath(*arg):
   return os.path.relpath(os.path.join(*arg),lib.app.workingDir)
 
+def allindir(dir):
+  return ' '.join( [ os.path.join(dir, file) for file in os.listdir(dir) ] )
+
 from lib.errorMessage  import errorMessage
 from lib.getHeaderInfo import getHeaderInfo
 from lib.getUserPath   import getUserPath
@@ -215,11 +218,11 @@ else:
       runCommand('rm -f masks_transformed/' + i.prefix + '_translated.mif')
 
 
-runCommand('mrmath input_transformed/* mean initial_template.mif')
+runCommand('mrmath ' + allindir('input_transformed') + ' mean initial_template.mif')
 current_template = 'initial_template.mif'
 
 if useMasks:
-  runCommand('mrmath masks_transformed/* min initial_template_mask.mif')
+  runCommand('mrmath ' + allindir('masks_transformed') + ' min initial_template_mask.mif')
   current_template_mask = 'initial_template_mask.mif'
 
 # Optimise template with linear registration
@@ -261,7 +264,7 @@ for level in range(0, len(linear_scales)):
                  output)
 
   # Here we ensure the template doesn't drift or scale
-  runCommand('transformcalc linear_transforms/*.txt average linear_transform_average.txt -force -quiet')
+  runCommand('transformcalc ' + allindir('linear_transforms') + ' average linear_transform_average.txt -force -quiet')
   if dorigid:
     runCommand('transformcalc linear_transform_average.txt rigid linear_transform_average.txt -force -quiet')
   runCommand('transformcalc linear_transform_average.txt invert linear_transform_average_inv.txt -force -quiet')
@@ -285,11 +288,11 @@ for level in range(0, len(linear_scales)):
                ' masks_transformed/' + i.prefix + '.mif' +
                ' -force')
 
-  runCommand ('mrmath input_transformed/* mean linear_template' + str(level) + '.mif')
+  runCommand ('mrmath ' + allindir('input_transformed') + ' mean linear_template' + str(level) + '.mif')
   current_template = 'linear_template' + str(level) + '.mif'
 
   if useMasks:
-    runCommand ('mrmath masks_transformed/* min linear_template_mask' + str(level) + '.mif')
+    runCommand ('mrmath ' + allindir('masks_transformed') + ' min linear_template_mask' + str(level) + '.mif')
     current_template_mask = 'linear_template_mask' + str(level) + '.mif'
 
 # Optimise the template with non-linear registration
@@ -339,11 +342,11 @@ for level in range(0, len(nl_scales)):
                  ' -interp nearest ' +
                  ' -force')
 
-  runCommand ('mrmath input_transformed/* mean nl_template' + str(level) + '.mif')
+  runCommand ('mrmath ' + allindir('input_transformed') + ' mean nl_template' + str(level) + '.mif')
   current_template = 'nl_template' + str(level) + '.mif'
   
   if useMasks:
-    runCommand ('mrmath masks_transformed/* min nl_template_mask' + str(level) + '.mif')
+    runCommand ('mrmath ' + allindir('masks_transformed') + ' min nl_template_mask' + str(level) + '.mif')
     current_template_mask = 'nl_template_mask' + str(level) + '.mif'
     
   if level < len(nl_scales) - 1:

